src/registry/emit.rs

```diff
@@ -0,0 +1,440 @@
+    let exporter = opentelemetry_otlp::SpanExporter::builder()
+        .with_tonic()
+        .with_endpoint("http://localhost:4317")
+        .build()?;
+    Ok(sdktrace::TracerProvider::builder()
+        .with_resource(Resource::new(vec![KeyValue::new("service.name", "weaver")])) // TODO meta semconv!
+        .with_batch_exporter(exporter, opentelemetry_sdk::runtime::Tokio)
+        .build())
+}
+
+/// Initialise a stdout exporter for debug
+fn init_stdout_tracer_provider() -> sdktrace::TracerProvider {
+    sdktrace::TracerProvider::builder()
+        .with_resource(Resource::new(vec![KeyValue::new("service.name", "weaver")])) // TODO meta semconv!
+        .with_simple_exporter(opentelemetry_stdout::SpanExporter::default())
+        .build()
+}
+
+/// Uses the global tracer_provider to emit a single trace for all the defined
+/// spans in the registry
+fn emit_trace_for_registry(registry: &ResolvedRegistry, registry_path: &str) {
+    let tracer = global::tracer("weaver");
+    // Start a parent span here and use this context to create child spans
+    tracer.in_span("weaver.emit", |cx| {
+        let span = cx.span();
+        span.set_attribute(KeyValue::new(
+            "weaver.registry_path", // TODO meta semconv!
+            registry_path.to_owned(),
+        ));
+
+        // Emit each span to the OTLP receiver.
+        for group in registry.groups.iter() {
+            if group.r#type == GroupType::Span {
+                let _span = tracer
+                    .span_builder(group.id.clone())
+                    .with_kind(otel_span_kind(group.span_kind.as_ref()))
+                    .with_attributes(group.attributes.iter().map(get_attribute_name_value))
+                    .start_with_context(&tracer, &cx);
+            }
+        }
+    });
+}
+
+/// Emit all spans in the resolved registry.
+pub(crate) fn command(
+    logger: impl Logger + Sync + Clone,
+    args: &RegistryEmitArgs,
+) -> Result<ExitDirectives, DiagnosticMessages> {
+    logger.log("Weaver Registry Emit");
+    logger.loading(&format!("Resolving registry `{}`", args.registry.registry));
+
+    let mut diag_msgs = DiagnosticMessages::empty();
+
+    let (registry, _) =
+        prepare_main_registry(&args.registry, &args.policy, logger.clone(), &mut diag_msgs)?;
+
+    logger.loading(&format!("Emitting registry `{}`", args.registry.registry));
+    let rt = tokio::runtime::Runtime::new().expect("failed to create tokio runtime");
+    rt.block_on(async {
+        let tracer_provider = if args.stdout {
+            logger.mute();
+            init_stdout_tracer_provider()
+        } else {
+            init_tracer_provider().expect("OTLP Tracer Provider must be created")
+        };
+        let _ = global::set_tracer_provider(tracer_provider.clone());
+
+        emit_trace_for_registry(&registry, &args.registry.registry.to_string());
+
+        global::shutdown_tracer_provider();
+    });
```

Comment:
Regarding the conversation with @jsuereth this morning, I think the creation of the Tokio runtime should be encapsulated in the `weaver_emit` crate mentioned earlier to avoid any async aspects at this level (even though it is honestly quite limited).

Reply:
I have done this now.